### PR TITLE
Add failing test for BN overflow issue

### PR DESCRIPTION
**Pull request by yvettep321**
https://github.com/MetaMask/metamask-extension/issues/7881

### CODE REVIEW

File: test/index.ts [hunk 1]
```diff
@@ -578,7 +578,7 @@ test('signedTypeData_v4', (t) => {
     domain: {
         name: 'Ether Mail',
         version: '1',
-        chainId: 1,
+        chainId: 9007199254740992,
```
Comment (mcmire) [inline test/index.ts:581]: Instead of changing this test can we add a new test? Otherwise the `try` part of the `try`/`catch` will never get hit.

**Comment by Gudahtt** [comment]
It seems like this was a mistake? If not, please explain your intent with creating this PR @yvettep321.

This commit is an old branch Finlay tested to create a repro for an extension bug.
